src/gui/socketapi.cpp

```diff
@@ -704,7 +704,10 @@ SocketApi::FileData SocketApi::FileData::get(const QString &localFile)
         return data;
 
     data.accountRelativePath = QDir(data.folder->remotePath()).filePath(data.folderRelativePath);
-
+    QString virtualFileExt = QStringLiteral(APPLICATION_DOTVIRTUALFILE_SUFFIX);
+    if (data.accountRelativePath.endsWith(virtualFileExt)) {
+        data.accountRelativePath.chop(virtualFileExt.size());
+    }
```

Comment:
It seems confusing to me to do this operation to the account-path but not to the folder-path. 

It's indeed likely that the account-path will be used in server queries and the like while the folder path wouldn't be - but one of them refering to the virtual file while the other doesn't might cause hard to spot issues.

Reply:
Right, in the future, we could decouple the two.
The problem is that I don't really want to recursively go into the folder to see if there are placeholders.

(altough maybe a SELECT WHERE type=4 AND IS_PREFIX_PATH_OF  might be simple enough)

Reply:
(Note: I removed the previous comment because it was not applying here.)

I think it make sense to have the server path relate to the file on the server.
Maybe I should specify it in the code comments.

Reply:
If you rename it serverPath that's fine